### DISCUSSION TITLE
New Peggy-based parser

### DISCUSSION THREAD

**Discussion by nichtich** [pg-format/pg-formatter#12] [Show and tell]
I've rewritten the PG format parser used in pgraph from scratch with [Peggy grammar](https://github.com/gbv/pgraph/blob/main/src/pg.pegjs) and learned a lot about parsers. The grammar differs from the grammar in this repository because it does not preserve formatting. Furthermore there are some aspects to be discussed, I'll open independent discussions for these.

**Comment by nichtich** [comment]
See https://github.com/gbv/pgraph/blob/main/docs/pg-grammar.md for railroad diagram generated from this grammar.

**Comment by hchiba1** [comment]
By the way, should the package-lock.json file of the `pgraph` repository be pushed?
(or specified in `.gitignore`?)

**Comment by hchiba1** [comment]
Is there good documentation describing pros and cons of using Peggy instead of PEG.js ?
* Pros: some new syntax sugar and performance improvements?
  * Maybe one of the most significant updates is `@` operator?
* No cons?
  * Maybe no reason to select PEG.js anymore.

**Reply by nichtich** [reply]
PEG.js is not maintained anymore, so Peggy is just the new name with some updates.

**Comment by hchiba1** [comment]
Saved some information here:
https://github.com/pg-format/pg-formatter/wiki
